### ISSUE TITLE
[@typescript-eslint/ban-types] Does not detect `{}` with a line break inside

### ISSUE THREAD

**Issue by HolgerJeromin** [typescript-eslint/typescript-eslint#4220]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/main/docs/linting/TROUBLESHOOTING.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/ban-types": "error"
  }
}
```

```TS
let foo: {
};
let bar:{};
```

**Expected Result**

Both expressions should report the same message:
> Don't use `{}` as a type. `{}` actually means "any non-nullish value". [...] (@typescript-eslint/ban-types)

**Actual Result**
The expression with the newline is not detected.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.4.0` |
| `@typescript-eslint/parser`        | `5.4.0` |
| `TypeScript`                       | `4.5.2` |
| `ESLint`                           | `7.32.0` |
